src/main/java/uk/gov/hmcts/darts/arm/service/impl/ArmResponseFilesProcessorImpl.java

```diff
@@ -0,0 +1,410 @@
+            }
+        } else {
+            log.warn("External object id {} checksum differs. Arm checksum: {} Object Checksum: {}",
+                     externalObjectDirectory.getId(),
+                     armResponseUploadFileRecord.getMd5(), objectChecksum
+            );
+            updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+        }
+
+    }
+
+    private UploadNewFileRecord readInputJson(ExternalObjectDirectoryEntity externalObjectDirectory, String input) {
+        UploadNewFileRecord uploadNewFileRecord = null;
+        ObjectRecordStatusEntity armResponseProcessingFailed = objectRecordStatusRepository.getReferenceById(ARM_RESPONSE_PROCESSING_FAILED.getId());
+        if (StringUtils.isNotEmpty(input)) {
+            String unescapedJson = StringEscapeUtils.unescapeJson(input);
+            try {
+                uploadNewFileRecord = objectMapper.readValue(unescapedJson, UploadNewFileRecord.class);
+            } catch (JsonMappingException e) {
+                log.error("Unable to map the upload record file input field");
+                updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+            } catch (JsonProcessingException e) {
+                log.error("Unable to parse the upload record file ");
+                updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+            }
+        } else {
+            log.warn("Unable to get the upload record file input field");
+            updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+        }
+        return uploadNewFileRecord;
+    }
+
+    private void updateExternalObjectDirectoryStatusAndVerificationAttempt(ExternalObjectDirectoryEntity externalObjectDirectory,
+                                                                           ObjectRecordStatusEntity objectRecordStatus) {
+        if (externalObjectDirectory.getVerificationAttempts() <= armDataManagementConfiguration.getMaxRetryAttempts()) {
+            int verificationAttempts = externalObjectDirectory.getVerificationAttempts() + 1;
+            externalObjectDirectory.setVerificationAttempts(verificationAttempts);
+            updateExternalObjectDirectory(externalObjectDirectory, objectRecordStatus);
+        } else {
+            ObjectRecordStatusEntity armResponseProcessingFailed = objectRecordStatusRepository.getReferenceById(ARM_RESPONSE_PROCESSING_FAILED.getId());
+            updateExternalObjectDirectory(externalObjectDirectory, armResponseProcessingFailed);
+        }
+    }
+
+    private void updateExternalObjectDirectory(ExternalObjectDirectoryEntity externalObjectDirectory, ObjectRecordStatusEntity objectRecordStatus) {
+        externalObjectDirectory.setStatus(objectRecordStatus);
+        externalObjectDirectory.setLastModifiedBy(userIdentity.getUserAccount());
+        externalObjectDirectory.setLastModifiedDateTime(OffsetDateTime.now());
```

Comment:
This is set automatically, and so doesn't need to be specified.

Reply:
In my testing we found that the automatic tasks don't update these values on modifications